src/client/metadataApiRetrieve.ts

```diff
@@ -175,7 +191,7 @@ export class MetadataApiRetrieve extends MetadataTransfer<
 
     await this.maybeSaveTempDirectory('source', components);
 
-    return new RetrieveResult(result, components);
+    return new RetrieveResult(result, components, this.components);
```

Comment:
suggestion: `components` and `this.components` is confusing. Could `components` be `componentsFromZip`

Reply:
especially with the RetrieveResult and the MetadataApiRetrieve classes being in the same file (because MetadataApiRetrieve has its own definition of `this.components`, which was [I think?] the componentSet that retrieve was called with)

Reply:
`components` == the `ComponentSet` retrieved from the org.  Calling `getSourceComponents()` on this comp set would yield every component in the zip tree container.
`this.components` == the `ComponentSet` used to build the request to send to the org.  Calling `getSourceComponents()` on this comp set yields only the components on the local file system (technically, the TreeContainer).  The difference between the 2 comp sets are the "Created" files.

I can rename to be more clear.

Reply:
Renaming would be a breaking change so I instead updated the constructor docs to state what `components` refers to.